// crud/index.js
const mysql = require('mysql')

class ClassCrud {
    
    constructor() {
        this.sql = mysql.createConnection({
            host        : 'localhost',
            user        : 'curso',    
            database    : 'celke',
            password    : '040303',
            port        : '3306'
        })
    }

    create(req, res) {
        this.sql.query(`
            INSERT INTO alunos
            VALUES 
            (?, ?);
        `, [req.body.name, req.body.dNasc])
        res.render('controllerForm')
    }

    read(req, res) {
        if (!req.params.Nome) {
            this.sql.query('SELECT * FROM alunos;', function (error, results, fields) {
                res.render('select', {data: results})            
            })
        } else {
            this.sql.query('SELECT * FROM alunos WHERE Nome like ?', [req.params.Nome], function(error, results, fields) {
                res.render('select', {data: results})
            })
        }
    }

    update(req, res) {
        this.sql.query('UPDATE alunos SET Nome = ?, DNasc = ? WHERE Nome like ?', [req.body.name, req.body.dNasc, req.body.param])
        res.send('<h1>Dados atualizados com sucesso!</h1>')
    }

    delete(req, res) {
        this.sql.query('DELETE FROM alunos WHERE Nome = ?', [req.params.Nome])
        res.render('deletar')
    }
}

module.exports = ClassCrud
